### ISSUE TITLE
taskonomy datasets does not contain mask_valid task.

### ISSUE THREAD

**Issue by FYtrace** [EPFL-VILAB/omnidata#4]
Thanks for your excellent work. I met some issues when running your code. Details as follows: 

- First. taskonomy datasets does not contain mask_valid task. 
https://github.com/EPFL-VILAB/omnidata-tools/blob/7b77a4fba2fbf55b22ed4af226125d5255e55330/omnidata_tools/torch/train_depth.py#L107

  https://github.com/EPFL-VILAB/omnidata-tools/blob/7b77a4fba2fbf55b22ed4af226125d5255e55330/omnidata_tools/torch/train_depth.py#L348

could I generate mask_valid according to gt_depth? sets 2**16-1 as invalid pixel?

- Second. pytorch_lightning throws `validation_step() missing 1 required positional argument: 'dataset_idx'`  error.
https://github.com/EPFL-VILAB/omnidata-tools/blob/7b77a4fba2fbf55b22ed4af226125d5255e55330/omnidata_tools/torch/train_depth.py#L192

Looking forward to your reply~

**Comment by Ainaz99** [comment]
Hi @FYtrace !

Yes, the invalid pixels have maximum depth value (`2**16 - 1`) and you can generate the masks just like you said.
And for your second question, I think it's a version issue. What pytorch_lightning version are you using?

**Comment by FYtrace** [comment]
Hi， @Ainaz99 , Thanks for your quick reply !

My pytorch_lightning version is 1.1.8. I also tried some other latest versions, but all failed. Now, I successfully ran by deleting argument 'dataset_idx'. Maybe I should merge the val sets into single one. 

Thanks for your wonderful work again. I will continue to follow your work and do something interesting.

**Comment by Ainaz99** [comment]
Yes, merging the validation sets can be a solution!

**Comment by icoz69** [comment]
hi, how do you download the ground truth depth data? I have only /rgb and /mask_valid downloaded, by running `omnitools.download rgb depth mask_valid --components replica --subset fullplus --dest ./omnidata_starter_dataset/ --connections_total 40 --agree_all --name XXX --email XXX`

**Comment by icoz69** [comment]
hi, how do you handle the second issue? i came across the same problem,

**Comment by puyiwen** [comment]
> 感谢您的出色工作。我在运行您的代码时遇到了一些问题。详情如下：
> 
> * 第一的。taskonomy 数据集不包含 mask_valid 任务。
>   https://github.com/EPFL-VILAB/omnidata-tools/blob/7b77a4fba2fbf55b22ed4af226125d5255e55330/omnidata_tools/torch/train_depth.py#L107
>   https://github.com/EPFL-VILAB/omnidata-tools/blob/7b77a4fba2fbf55b22ed4af226125d5255e55330/omnidata_tools/torch/train_depth.py#L348
> 
> 我可以根据 gt_depth 生成 mask_valid 吗？将 2**16-1 设置为无效像素？
> 
> * 第二。pytorch_lightning 抛出`validation_step() missing 1 required positional argument: 'dataset_idx'`  错误。
>   https://github.com/EPFL-VILAB/omnidata-tools/blob/7b77a4fba2fbf55b22ed4af226125d5255e55330/omnidata_tools/torch/train_depth.py#L192
> 
> 期待您的回复~

Hi,how can I set the  depth value (2**16 - 1) as  invalid pixels?Like set the depth value as 0 which is (2**16 -1)?

**Comment by wtpro** [comment]
> Hi， @Ainaz99 , Thanks for your quick reply !
> 
> My pytorch_lightning version is 1.1.8. I also tried some other latest versions, but all failed. Now, I successfully ran by deleting argument 'dataset_idx'. Maybe I should merge the val sets into single one.
> 
> Thanks for your wonderful work again. I will continue to follow your work and do something interesting.

Hi, can you elaborate on how to solve the dataset_idx issue? Simply by deleting the argument did not solve the issue at all because validation_step returns a variable that depends on it.
